modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/display/context/ContentPageEditorDisplayContext.java

```diff
@@ -708,18 +708,28 @@ public Map<String, Object> getEditorContext() throws Exception {
 						return styleBookEntryERC;
 					}
 
-					if (Validator.isNull(styleBookEntryERC)) {
+					if (Validator.isNotNull(styleBookEntryERC)) {
 						StyleBookEntry defaultStyleBookEntry =
 							DefaultStyleBookEntryUtil.getDefaultStyleBookEntry(
 								layout);
 
-						if (defaultStyleBookEntry != null) {
-							return defaultStyleBookEntry.
-								getExternalReferenceCode();
+						StyleBookEntry styleBookEntry =
+							_styleBookEntryLocalService.
+								fetchStyleBookEntryByExternalReferenceCode(
+									styleBookEntryERC,
+									_staging.getLiveGroupId(
+										layout.getGroupId()));
+
+						if ((styleBookEntry != null) &&
+							Objects.equals(
+								defaultStyleBookEntry.getThemeId(),
```

Comment:
This seems like an odd way to get the themeId assigned to the layout. Wouldn't it be better to use `frontendTokenDefinitionRegistry.getFrontendTokenDefinition(layout)` and then get the themeId from the frontendTokenDefinition?